radar/exporter/exporters.py

```diff
@@ -357,96 +357,164 @@ def setup(self):
-                column("biopsy_diagnosis_label"),
-                d("comments", anonymised_getter=None)
-            ]
-        )           
+            self._columns.extend(
+                [
+                    column("ins_diagnosis"), #27
+                    column('ins_diagnosis_date'),
+                    column('ins_biopsy_diagnosis'),
+                    column('ins_biopsy_diagnosis_label'),
+                    column('ins_diagnosis_comments') #31                                  
+                ]
+            )
         self._columns.extend(get_meta_columns(self.config))
         self._query = queries.get_patient_diagnoses(self.config)
-        self._primary = queries.get_primary_diagnoses(self.config)        
+        self._primary = queries.get_primary_diagnoses(self.config)
 
 @register("primary-diagnoses")
 class PrimaryDiagnosisExporter(DiagnosisExporter):
-    
+        
     def get_rows(self):
         headers = [col[0] for col in self._columns]
         yield headers
 
-        self.ins_data = {
-            'dia_name': '',
-            'dia_date': ''
-        }
-        self.row = []        
-
-        for result in self._query:
-            # if not being run for the first time and new patient
-            if len(self.row) > 1 and result.patient.id != self.row[1]:
-                # if the export is being run for INS
-                if 'INS diagnosis' in headers:
-                    self.row[8] = self.ins_data['dia_name']
-                    self.ins_data['dia_name'] = ''
-                    self.row[9] = self.ins_data['dia_date']
-                    self.ins_data['dia_date'] = ''
-                    
-                                
-                yield self.row
-                self.row = []
-                self.makeRow(result)
-
-            else:
-                self.makeRow(result)
-
-
-    def makeRow(self, result):
-        diagnosis = result.diagnosis
-            
-        if diagnosis and diagnosis.groups:
-                for group in diagnosis.group_diagnoses:
-                    if group.group.code == 'INS' and group.type.value == 'PRIMARY':
-                        if not self.ins_data['dia_date'] or self.ins_data['dia_date'] < result.from_date:
-                            self.ins_data['dia_name'] = diagnosis.name 
-                            self.ins_data['dia_date'] = result.from_date
+        # Differnet format for Nurture INS patients
+        if self.config['patient_group'].code == 'NURTUREINS':
+
+            patient_id = 0            
+            row = []
+
+            # Dictonary containing INS data to be added to Nurture INS data
+            self.ins_data = {
+                'pat_id': '',
+                'ins_dia_name': '',
+                'ins_dia_date': '',
+                'ins_biopsy_dia': '',
+                'ins_biopsy_dia_label': '',
+                'ins_biopsy_dia_comments': ''                
+            }
+
+            for result in self._query:
```

Comment:
Will add a result comment here

Reply:
The problem with leaving it there is:

- `for result in self._query:` is pretty self documenting, so not really needed here as your wonderful variable names do the trick.

- It still would not explain what the `result` argument in the method signature `def getInsData(self, result, diagnosis):` if the context changed.

So maybe a doc string in the method?